FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataFilter/ContainerRegistryFilterProvider.cs

```diff
@@ -62,7 +63,13 @@ public async Task<FilterConfiguration> GetFilterAsync(CancellationToken cancella
             {
                 AcrClient client = new AcrClient(imageInfo.Registry, accessToken);
                 var filterConfigurationProvider = new OciArtifactProvider(imageInfo, client);
-                var acrImage = await filterConfigurationProvider.GetOciArtifactAsync(cancellationToken);
+                var acrImage = await Policy
+                  .Handle<TemplateManagementException>()
+                  .RetryAsync(3, onRetry: (exception, retryCount) =>
+                  {
+                      _logger.LogWarning(exception, "Failed to get image from Container Registry. Retry {RetryCount}.", retryCount);
```

Comment:
Maybe we can add the image reference to the logs? like `Failed to get image {reference} from container registry.

Reply:
Added, thanks.